fix for issue #42611

nhavens opened the pull request:
### What does this PR do?
Fixes return values from `selinux.boolean` state.

### What issues does this PR fix or reference?
#42611 - selinux.boolean state does not return changes

### Previous Behavior
1. `selinux.boolean` does not return a list of changes, and therefor cannot be used by requisites listening for any changes.
2. `selinux.boolean` returns `True` rather than `False` in live mode if it attempted changes and there was an error.

### New Behavior
All the above issues have been corrected.

Commit:
fix for issue #42611

salt-jenkins commented:
@nhavens, thanks for your PR! By analyzing the history of the files in this pull request, we identified @terminalmage, @thatch45 and @t0fik to be potential reviewers.

Commit:
fix test_selinux for issue #42611

- expect change from selinux.boolean state if it makes changes
- result = False if there's an error making changes

garethgreenaway committed:
Merge branch 'develop' into develop

garethgreenaway committed:
Merge branch 'develop' into develop

t0fik commented:
@nhavens It would be nice to backport this fix to 2017.7 branch

nhavens committed:
Merge branch 'develop' into develop

nhavens commented:
@t0fik, I just submitted PR #42657 to back-port the fix to 2017.7.

I hope I did it right. I'm new to git and to github.

nhavens committed:
Merge branch 'develop' into develop
